File: mesrapp-backend/mesrapp/src/main/java/com/example/mesrapp/model/Match.java
package com.example.mesrapp.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "matches")
public class Match {
    @Id
    @Column(nullable = false, unique = true)
    private String spotiId;

    @Column(nullable = false)
    private String musicId;
    
    @Column(nullable = false)
    private String artistId;
}

File: mesrapp-frontend/mesrapp/app/src/main/java/com/example/mesrapp/models/Match.kt
package com.example.mesrapp.models

import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity
data class Match (
    @ColumnInfo(name = "spotiId", index = true) val spotiId: String,
    @ColumnInfo(name = "musicId") val musicId: String,
    @ColumnInfo(name = "artistId") val artistId: String,
) {
    constructor(spotiId: String, musicId: String)
            : this(spotiId = spotiId, musicId = musicId, artistId = "")
}